HomeWork/src/main/java/generics/Box.java

```diff
@@ -0,0 +1,101 @@
+package generics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Box<T extends Fruit> {
+
+    private List<Fruit> fruitBox = new ArrayList<>();
+    private T fruit;
+
+    public Box(T fruit) {
+        this.fruit = fruit;
+    }
+
+
+    public T getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(T fruit) {
+        this.fruit = fruit;
+    }
+
+    public List<Fruit> getFruitBox() {
+        return fruitBox;
+    }
+
+    public void setFruitBox(List<Fruit> fruitBox) {
+        this.fruitBox = fruitBox;
+    }
+
+    public <O extends T> void add(O obj) {
+        Objects.requireNonNull(obj);
+        this.fruitBox.add(obj);
+    }
+
+    public boolean addAllFruit(List<? extends Fruit> fruitList) {
+        if (fruitList.isEmpty()) {
+            throw new NullPointerException("An empty list is passed to the method argument");
+        }
```

Comment:
You are checking whether the `fruitList` is empty, which means it's not null but has no elements, and if so, you can't throw NullPointer, it is not correct. Get rid of this check.